### ISSUE TITLE
Memcached incompatability

### ISSUE THREAD

**Issue by zodiac** [facebook/hhvm#6949]
### HHVM Version

3.10.1

### Standalone code, or other way to reproduce the problem

```
<?php

$mem = new Memcached(...);
$mem->addServers(...);

$mem->setOptions([
	Memcached::OPT_BINARY_PROTOCOL => 1
]);

$key = generateRandomString();

var_dump($mem->increment($key, 1, 0));
var_dump($mem->getResultCode());
var_dump($mem->increment($key, 1, 0));
var_dump($mem->getResultCode());
var_dump($mem->get($key));

$key2 = generateRandomString();

var_dump($mem->increment($key2, 1));
var_dump($mem->getResultCode());
var_dump($mem->increment($key2, 1));
var_dump($mem->getResultCode());
var_dump($mem->get($key));

$key3 = generateRandomString();

var_dump($mem->increment($key3));
var_dump($mem->getResultCode());
var_dump($mem->increment($key3));
var_dump($mem->getResultCode());
var_dump($mem->get($key));
```

### Expected result

Result when run with php 5.5.33, pecl_memcached 2.2.0, libmemcached 1.0.16

```
int(0)
int(0)
int(1)
int(0)
string(1) "1"
bool(false)
int(16)
bool(false)
int(16)
string(1) "1"
bool(false)
int(16)
bool(false)
int(16)
string(1) "1"
```

### Actual result

Result when running on hhvm 3.10.1

```
int(0)
int(0)
int(1)
int(0)
string(1) "1"
int(-1)
int(21)
int(0)
int(0)
string(1) "1"
int(-1)
int(21)
int(0)
int(0)
string(1) "1"
```

**Comment by fredemmott** [comment]
3.10 is insupported; are you able to reproduce the problem on 3.12.1?

**Comment by zodiac** [comment]
I could reproduce the problem off the nightly build (3.13.0-dev).

The problem seems to me to be a design bug - pecl_memcached's increment behaves differently if you omit the second and third arguments, even though they are documented as having default values.

**Comment by Orvid** [comment]
I believe this was fixed in PR #7000.
